website/docs/d/is_security_group_rule.html.markdown

```diff
@@ -0,0 +1,103 @@
+---
+layout: "ibm"
+page_title: "IBM : security_group_rule"
+sidebar_current: "docs-ibm-resource-is-security-group-rule"
+description: |-
+  Manages IBM Security Group Rule.
+---
+
+# ibm\_is_security_group_rule
+
+Provides a security group rule data source. This allows security group rule to be read.
+
+
+## Example Usage
+
+In the following example, you can create a different types of protocol rules `ALL`, `ICMP`, `UDP` and `TCP`.
+
+```hcl
+resource "ibm_is_vpc" "testacc_vpc" {
+  name = "test"
+}
+
+resource "ibm_is_security_group" "testacc_security_group" {
+  name = "test"
+  vpc  = ibm_is_vpc.testacc_vpc.id
+}
+
+resource "ibm_is_security_group_rule" "testacc_security_group_rule_all" {
+  group     = ibm_is_security_group.testacc_security_group.id
+  direction = "inbound"
+  remote    = "127.0.0.1"
+}
+
+resource "ibm_is_security_group_rule" "testacc_security_group_rule_icmp" {
+  group     = ibm_is_security_group.testacc_security_group.id
+  direction = "inbound"
+  remote    = "127.0.0.1"
+  icmp {
+    code = 20
+    type = 30
+  }
+}
+
+resource "ibm_is_security_group_rule" "testacc_security_group_rule_udp" {
+  group     = ibm_is_security_group.testacc_security_group.id
+  direction = "inbound"
+  remote    = "127.0.0.1"
+  udp {
+    port_min = 805
+    port_max = 807
+  }
+}
+
+resource "ibm_is_security_group_rule" "testacc_security_group_rule_tcp" {
+  group     = ibm_is_security_group.testacc_security_group.id
+  direction = "egress"
+  remote    = "127.0.0.1"
+  tcp {
+    port_min = 8080
+    port_max = 8080
+  }
+}
+
+data "ibm_is_security_group_rule" "sg1_rule" {
+  name = ibm_is_security_group.testacc_security_group.name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `group` - (Required, Forces new resource, string) The security group id.
```

Comment:
why all these attributes are under argment its the datasource not the resource...only name is required parameter

remaining all are computed